om4labs/diags/seaice/seaice.py

```diff
@@ -0,0 +1,449 @@
+    )
+
+    # Create annual cycle climatology
+    model = averagers.annual_cycle(ds, "CN")
+    obs = averagers.annual_cycle(dobs, "sic")
+
+    # Calculate area and extent
+    if region == "nh":
+        area_model = np.where(ds["GEOLAT"] > 0.0, model * ds.AREA, 0.0)
+        ext_model = np.where((model > 0.15) & (ds["GEOLAT"] > 0.0), ds.AREA, 0.0)
+    elif region == "sh":
+        area_model = np.where(ds["GEOLAT"] < 0.0, model * ds.AREA, 0.0)
+        ext_model = np.where((model > 0.15) & (ds["GEOLAT"] < 0.0), ds.AREA, 0.0)
+    else:
+        raise ValueError(f"Unknown region {region}. Option are nh or sh")
+    area_model = area_model.sum(axis=(-2, -1)) * 1.0e-12
+    ext_model = ext_model.sum(axis=(-2, -1)) * 1.0e-12
+
+    area_obs = obs * dobs.areacello
+    area_obs = area_obs.sum(axis=(-2, -1)) * 1.0e-12
+    ext_obs = np.where(obs > 0.15, dobs.areacello, 0.0)
+    ext_obs = ext_obs.sum(axis=(-2, -1)) * 1.0e-12
+
+    # Add back in the 2D coordinates
+    model["GEOLON"] = ds["GEOLON"]
+    model["GEOLAT"] = ds["GEOLAT"]
+    model = model.rename({"GEOLON": "lon", "GEOLAT": "lat"})
+    obs["lon"] = dobs["lon"]
+    obs["lat"] = dobs["lat"]
+
+    # Close original files to free up memory
+    ds.close()
+    dobs.close()
+
+    # Regrid the observations to the model grid (for plotting)
+    obs_r = regrid.curv_to_curv(obs, model, reuse_weights=False)
+
+    return (
+        model,
+        obs_r,
+        ext_model,
+        ext_obs,
+        area_model,
+        area_obs,
+        valid_mask,
+        modtime,
+        obstime,
+    )
+
+
+def plot_annual_cycle(ax, _mod, _obs, roll=0):
+    """Creates an anual cycle subplot panel"""
+    _months = np.array(["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"])
+
+    # Shift annual cycle
+    _months = np.roll(_months, roll)
+    _obs = np.roll(_obs, roll)
+    _mod = np.roll(_mod, roll)
+
+    # Add wrap-around point
+    _months = np.append(_months, _months[0])
+    _obs = np.append(_obs, _obs[0])
+    _mod = np.append(_mod, _mod[0])
+
+    # Plot the data
+    t = np.arange(0, len(_mod))
+    ax.plot(t, _mod, "r", label="model")
+    ax.plot(t, _obs, "k", label="obs")
+
+    # Format the plot
+    ax.set_xticks(np.arange(0, len(_obs)))
+    ax.set_xticklabels(_months)
+    plt.legend()
+
+
+def plot_map_panel(
+    ax,
+    x,
+    y,
+    plotdata,
+    cmap="Blues_r",
+    vmin=0.01,
+    vmax=100.0,
+    extent=[-180, 180, 50, 90],
+    contour=True,
+):
+    """Function to plot a map-based sea ice panel"""
+    cmap = plt.get_cmap(cmap)
+    cmap.set_bad(color="#555555", alpha=1)
+    ax.set_extent(extent, ccrs.PlateCarree())
+    cb = ax.pcolormesh(
+        x, y, plotdata, transform=ccrs.PlateCarree(), cmap=cmap, vmin=vmin, vmax=vmax
+    )
+    if contour is True:
+        cs2 = ax.contour(
+            x, y, plotdata, transform=ccrs.PlateCarree(), levels=[15.0], colors=["r"]
+        )
+
+
+def return_month_index(month):
```

Comment:
there must be a way to use months directly and not use index

Reply:
Removed this function in db9598e